src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -340,13 +340,15 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService, TextAs
             textBlockService.computeTextBlocksForSubmissionBasedOnSyntax(textSubmission);
         }
 
-        Result result;
+        Result result = null;
         if (!Boolean.TRUE.equals(exampleSubmission.isUsedForTutorial()) || authCheckService.isAtLeastInstructorForExercise(textExercise, user)) {
             result = textSubmission.getLatestResult();
-            final List<Feedback> assessments = textAssessmentService.getAssessmentsForResult(result);
-            result.setFeedbacks(assessments);
+            if (result != null) {
+                final List<Feedback> assessments = textAssessmentService.getAssessmentsForResult(result);
+                result.setFeedbacks(assessments);
+            }
         }
-        else {
+        if (result != null) {
```

Comment:
```suggestion
        if (result == null) {
```
This fallback value should only be applied if the call before returned null, otherwise the fetched feedbacks would be overwritten.

Reply:
yes, you are right. Thanks for the good catch!